tests/kas_messaging.spec.ts

```diff
@@ -0,0 +1,110 @@
+
+const testInstanceName = 'test-instance-messaging';
+const testTopicName = `test-topic-name`;
+const testServiceAccountName = 'test-service-account';
+const testMessageKey = 'key';
+const consumerGroupId = 'test-consumer-group';
+const expectedMessageCount = 100;
+let credentials;
+let bootstrap: string;
+
+test.beforeEach(async ({ page }) => {
+  test.skip(config.startingPage === config.startingPageDefault);
+
+  await login(page);
+
+  await navigateToKafkaList(page);
+
+  await expect(page.getByRole('button', { name: 'Create Kafka instance' })).toBeVisible();
+
+  if ((await page.getByText(testInstanceName).count()) > 0 && (await page.locator('tr').count()) === 2) {
+    // Test instance present, nothing to do!
+  } else {
+    await page.waitForSelector('[role=progressbar]', {
+      state: 'detached',
+      timeout: config.kafkaInstanceCreationTimeout
+    });
+
+    for (const el of await page.locator(`tr >> a`).elementHandles()) {
+      const name = await el.textContent();
+
+      if (name !== testInstanceName) {
+        await deleteKafkaInstance(page, name);
+      }
+    }
+
+    if ((await page.getByText(testInstanceName).count()) === 0) {
+      await createKafkaInstance(page, testInstanceName);
+    }
+  }
+
+  await navigateToKafkaTopicsList(page, testInstanceName);
+  // Do not create topic if it already exists
+  await expect(page.getByText('Create topic')).toHaveCount(1);
+  if ((await page.locator('a', { hasText: testTopicName }).count()) === 0) {
+    await createKafkaTopic(page, testTopicName);
+  }
+
+  await navigateToSAList(page);
+  await expect(page.getByText('Create service account')).toHaveCount(1);
+  if ((await page.locator('tr', { hasText: testServiceAccountName }).count()) !== 0) {
+    await deleteServiceAccount(page, testServiceAccountName);
+  }
+  credentials = await createServiceAccount(page, testServiceAccountName);
+  bootstrap = await getBootstrapUrl(page, testInstanceName);
+
+  await navigateToAccess(page, testInstanceName);
+  await grantProducerAccess(page, credentials.clientID, testTopicName);
+
+  // Producer 100 messages
+  const producer = new KafkaProducer(bootstrap, credentials.clientID, credentials.clientSecret);
+  const producerResponse = await producer.produceMessages(testTopicName, expectedMessageCount, testMessageKey);
+  expect(producerResponse === true).toBeTruthy();
+});
+
+// test_6messages.py generate_messages_to_topic
+test('Consume messages from topic', async ({ page }) => {
+  await navigateToAccess(page, testInstanceName);
+  await grantConsumerAccess(page, credentials.clientID, testTopicName, consumerGroupId);
+
+  // Consume 100 messages
+  const consumer = new KafkaConsumer(bootstrap, consumerGroupId, credentials.clientID, credentials.clientSecret);
+  const consumerResponse = await consumer.consumeMessages(testTopicName, expectedMessageCount);
+  expect(consumerResponse).toBeGreaterThanOrEqual(expectedMessageCount);
```

Comment:
I still don't get why, if we are producing 100 messages we might be receiving more than that ... is it a wrong configuration of the acknowledgment?
This should be `toBeEqual` I think ...